Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -292,7 +292,15 @@ def bottle_formula(f)
 
         cd cellar do
           sudo_purge
-          safe_system "tar", "cf", tar_path, "#{f.name}/#{f.pkg_version}"
+          keg_root = "#{f.name}/#{f.pkg_version}"
+          safe_system "tar", "cf", tar_path,
+                      # Put the two metafiles at the start of the tarball,
+                      # before listing the actual root of the keg.
+                      # This ensures that even a partial tarball download
+                      # should contain these two files.
+                      "#{keg_root}/INSTALL_RECEIPT.json",
+                      "#{keg_root}/#{keg.relative_formula_path}",
+                      keg_root
```

Comment:
Check with `tar t` that the file `INSTALL_RECEIPT.json` and formula don't get included in the tarball twice. I'd hope not… but I'm paranoid.

Reply:
Please check the behaviour of both GNU tar and macOS tar.